src/rgw/driver/sfs/bucket.cc

```diff
@@ -534,11 +534,39 @@ int SFSBucket::abort_multiparts(
   return sfs::SFSMultipartUploadV2::abort_multiparts(dpp, store, this);
 }
 
+/**
+ * @brief Refresh this bucket's rep with the state obtained from the store.
+ *        Indeed it can happen that the rep of this bucket is obsolete due to
+ *        concurrent threads updating metadata using their own SFSBucket instance.
+ */
 int SFSBucket::try_refresh_info(
     const DoutPrefixProvider* dpp, ceph::real_time* /*pmtime*/
 ) {
-  ldpp_dout(dpp, 10) << __func__ << ": TODO" << dendl;
-  return -ENOTSUP;
+  auto bref = store->get_bucket_ref(get_name());
+  // maybe excessively safe approach: instead of directly assign the returned
+  // value to this.bucket let's check first if bref is valid.
+  if (!bref) {
+    lsfs_dout(dpp, 0) << fmt::format("no such bucket! {}", get_name()) << dendl;
+    return -ERR_NO_SUCH_BUCKET;
+  }
+  bucket = bref;
+
+  // update views like we do in the constructor
+
+  // info view
+  get_info() = bucket->get_info();
```

Comment:
yup, this is ugly ... Unfortunately I don't know of to avoid this due to the lack of a `set_info()`.
maybe this is a little better:

```
auto &info = get_info();
info = bucket->get_info();
```
?

Reply:
I dunno... Thinking about it some more, I'm inclined to just leave it as-is, given the `get_info() = whatever` form is used already in several other places:

```
> git grep 'get_info() = [a-z]' src/rgw
src/rgw/driver/rados/rgw_sal_rados.cc:  bucket->get_info() = info;
src/rgw/driver/rados/rgw_user.cc:  user->get_info() = user_info;
src/rgw/rgw_sal_daos.cc:    bucket->get_info() = info;
src/rgw/rgw_sal_daos.cc:    bucket->get_info() = info;
src/rgw/rgw_sal_dbstore.cc:    bucket->get_info() = info;
src/rgw/rgw_sal_motr.cc:    bucket->get_info() = info;
src/rgw/rgw_sal_motr.cc:    // bucket->get_info() = info;
```

Reply:
ok, let's keep that way.